opts/envdir.go

```diff
@@ -0,0 +1,56 @@
+package opts
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+/*
+Read in a directory with files acting as environment variables.
+*/
+func ParseEnvDir(directory string) ([]string, error) {
+	dir, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return []string{}, err
+	}
+
+	lines := []string{}
+	for _, info := range dir {
+		variable := info.Name()
+		// ignore directories and hidden files
+		if !info.IsDir() && variable[0] != '.' {
+			if strings.ContainsAny(variable, whiteSpaces) {
```

Comment:
> I think we need to check for more than just spaces and tabs:
Variables set by the user must have a name consisting solely of alphabetics, numerics, and underscores - the first of which must not be numeric.

If we want to be more restrictive, that's fine, but I'd suggest we do this elsewhere and cover all the cases of setting environment variables in Docker, not just in envdir.

fwiw, I don't think that's a requirement:
```
$ docker run --rm -e '~0SERVICE_NA-M E$&!@#=foo' -it ubuntu:14.04 env | grep SERVICE
~0SERVICE_NA-M E$&!@#=foo
```